### PR TITLE
Feat/imgix product custom attribute implementation

### PR DESCRIPTION
**Pull request by syedbadshah-toptal**
product custom attribute implementation completed.
system objects file added in meta data.
unit test added.

### CODE REVIEW

File: cartridges/int_imgix_products_sfra/cartridge/models/product/productImages.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    }, this);
+}
+
+/**
+ * Append image URL
+ * @param {String} imageUrl - Image Url
+ * @param {Number} sourceWidth - Source width
+ * @returns {String} Image Url
+ */
+function appendImageUrl(imageUrl, sourceWidth) {
```
Comment (luqven) [inline cartridges/int_imgix_products_sfra/cartridge/models/product/productImages.js:81]: In my opinion, this function should really be called `appendSourceWidth`. We're not appending anything else to the `imageUrl`.

```suggestion
function appendSourceWidth(imageUrl, sourceWidth) {
```

File: metadata/imgix_data/meta/system-objecttype-extensions.xml [hunk 2]
```diff
@@ -17,13 +17,43 @@
                 <mandatory-flag>false</mandatory-flag>
                 <externally-managed-flag>false</externally-managed-flag>
                 <min-length>0</min-length>
+            </attribute-definition>
+            <attribute-definition attribute-id="imgixBaseURL">
+                <display-name xml:lang="x-default">Imgix Base URL</display-name>
```
Comment (luqven) [inline metadata/imgix_data/meta/system-objecttype-extensions.xml:22]:
```suggestion
                <display-name xml:lang="x-default">imgix Base URL</display-name>
```

File: metadata/imgix_data/meta/system-objecttype-extensions.xml [hunk 3]
```diff
@@ -17,13 +17,43 @@
...
                 <attribute attribute-id="imgixapiKey"/>
                 <attribute attribute-id="imgixDefaultParams"/>
+                <attribute attribute-id="imgixBaseURL"/>
+            </attribute-group>
+        </group-definitions>
+    </type-extension>
+    <type-extension type-id="Product">
+        <custom-attribute-definitions>
+            <attribute-definition attribute-id="imgixData">
+                <display-name xml:lang="x-default">Imgix Images</display-name>
```
Comment (luqven) [inline metadata/imgix_data/meta/system-objecttype-extensions.xml:42]:
```suggestion
                <display-name xml:lang="x-default">imgix Images</display-name>
```

File: metadata/imgix_data/meta/system-objecttype-extensions.xml [hunk 4]
```diff
@@ -17,13 +17,43 @@
...
+                <mandatory-flag>false</mandatory-flag>
+                <visible-flag>false</visible-flag>
+                <externally-managed-flag>false</externally-managed-flag>
+                <order-required-flag>false</order-required-flag>
+                <externally-defined-flag>false</externally-defined-flag>
+            </attribute-definition>
+        </custom-attribute-definitions>
+        <group-definitions>
+            <attribute-group group-id="imgix">
+                <display-name xml:lang="x-default">Imgix</display-name>
```
Comment (luqven) [inline metadata/imgix_data/meta/system-objecttype-extensions.xml:55]:
```suggestion
                <display-name xml:lang="x-default">imgix</display-name>
```

File: metadata/imgix_data/meta/system-objecttype-extensions.xml (hunk 2 above)
Comment (syedbadshah-toptal) [reply metadata/imgix_data/meta/system-objecttype-extensions.xml:22]: Usually names are starting with capital alphabet. These will be visible on custom preferences. While keys usually created in camelCase notation.

File: metadata/imgix_data/meta/system-objecttype-extensions.xml (hunk 4 above)
Comment (syedbadshah-toptal) [reply metadata/imgix_data/meta/system-objecttype-extensions.xml:55]: Usually names are starting with capital alphabet. These will be visible on custom preferences. While keys usually created in camelCase notation.

File: cartridges/int_imgix_products_sfra/cartridge/models/product/productImages.js (hunk 1 above)
Comment (syedbadshah-toptal) [reply cartridges/int_imgix_products_sfra/cartridge/models/product/productImages.js]: I've updated the function name.

File: metadata/imgix_data/meta/system-objecttype-extensions.xml (hunk 2 above)
Comment (luqven) [reply metadata/imgix_data/meta/system-objecttype-extensions.xml:22]: Given that these are `display-names` and not keys I think we should follow the company style guide and make this `imgix Base URL`. I defer to @frederickfogerty as to whether this is _necessary_ or not.

File: metadata/imgix_data/meta/system-objecttype-extensions.xml (hunk 3 above)
Comment (luqven) [reply metadata/imgix_data/meta/system-objecttype-extensions.xml:42]: Same as my previous comment above.

File: metadata/imgix_data/meta/system-objecttype-extensions.xml (hunk 4 above)
Comment (luqven) [reply metadata/imgix_data/meta/system-objecttype-extensions.xml:55]: Same as my previous comment above.

File: metadata/imgix_data/meta/system-objecttype-extensions.xml (hunk 2 above)
Comment (frederickfogerty) [reply metadata/imgix_data/meta/system-objecttype-extensions.xml:22]: Second what Luis said - we should follow the company style guide which is to have imgix lowercase wherever possible

File: metadata/imgix_data/meta/system-objecttype-extensions.xml (hunk 3 above)
Comment (frederickfogerty) [reply metadata/imgix_data/meta/system-objecttype-extensions.xml:42]: Same as my previous reply to the previous comment about the previous issue

File: metadata/imgix_data/meta/system-objecttype-extensions.xml (hunk 4 above)
Comment (frederickfogerty) [reply metadata/imgix_data/meta/system-objecttype-extensions.xml:55]: Same as the previous reply to my previous reply about the comment previous to the previous comment about the previous suggestion previous to the previous suggestion

Comment (syedbadshah-toptal) [reply metadata/imgix_data/meta/system-objecttype-extensions.xml]: @frederickfogerty @luqven I've updated the display name as suggestion, thanks.

File: metadata/imgix_data/meta/system-objecttype-extensions.xml (hunk 3 above)
Comment (syedbadshah-toptal) [reply metadata/imgix_data/meta/system-objecttype-extensions.xml]: @frederickfogerty @luqven I've updated the display name as suggestion, thanks.

File: metadata/imgix_data/meta/system-objecttype-extensions.xml (hunk 2 above)
Comment (syedbadshah-toptal) [reply metadata/imgix_data/meta/system-objecttype-extensions.xml]: @frederickfogerty @luqven I've updated the display name as suggestion, thanks.

File: metadata/imgix_data/meta/system-objecttype-extensions.xml [hunk 5]
```diff
@@ -17,13 +17,43 @@
...
+                <description xml:lang="x-default">Include Protocol Eg: http:// or https://</description>
+                <type>string</type>
+                <mandatory-flag>false</mandatory-flag>
+                <externally-managed-flag>false</externally-managed-flag>
+                <min-length>0</min-length>
             </attribute-definition>
 		</custom-attribute-definitions>
         <group-definitions>
             <attribute-group group-id="ImgixAttributes">
                 <display-name xml:lang="x-default">IMGIX Attributes</display-name>
```
Comment (luqven) [inline metadata/imgix_data/meta/system-objecttype-extensions.xml:32]:
```suggestion
                <display-name xml:lang="x-default">imgix Attributes</display-name>
```

File: metadata/imgix_data/meta/system-objecttype-extensions.xml [hunk 6]
```diff
@@ -17,13 +17,43 @@
...
+                <display-name xml:lang="x-default">imgix Base URL</display-name>
+                <description xml:lang="x-default">Include Protocol Eg: http:// or https://</description>
+                <type>string</type>
+                <mandatory-flag>false</mandatory-flag>
+                <externally-managed-flag>false</externally-managed-flag>
+                <min-length>0</min-length>
             </attribute-definition>
 		</custom-attribute-definitions>
         <group-definitions>
             <attribute-group group-id="ImgixAttributes">
```
Comment (luqven) [inline metadata/imgix_data/meta/system-objecttype-extensions.xml:31]:
```suggestion
            <attribute-group group-id="imgixAttributes">
```

**Review by luqven — APPROVED** [review]
@syedbadshah-toptal looks like we missed two 😬 . Once those are changed this LGTM.

File: metadata/imgix_data/meta/system-objecttype-extensions.xml (hunk 5 above)
Comment (syedbadshah-toptal) [reply metadata/imgix_data/meta/system-objecttype-extensions.xml]: @luqven I've just updated as suggestion, thanks.

**Comment by frederickfogerty** [comment]
Closing this as it was implemented by #165

**Pull request closed without merging**